### PR TITLE
fix: Ignore undeletable messages in channel.purge; fix rate-limits issue. 

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

see title.

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [ ] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [x] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/client/context.py [hunk 1]
```diff
@@ -476,7 +476,7 @@ async def defer(self, ephemeral: Optional[bool] = False) -> None:
         """
         if not self.responded:
             self.deferred = True
-            _ephemeral: int = (1 << 6) if ephemeral else 0
+            _ephemeral: int = MessageFlags.EPHEMERAL.value if ephemeral else MessageFlags(0)
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
            _ephemeral: int = MessageFlags.EPHEMERAL.value if ephemeral else 0
```

File: interactions/api/models/message.py [hunk 2]
```diff
@@ -75,6 +76,15 @@ class MessageType(IntEnum):
     CONTEXT_MENU_COMMAND = 23
     AUTO_MODERATION_ACTION = 24
 
+    @staticmethod
+    def not_deletable() -> List[int]:
```
Comment (Damego) [inline interactions/api/models/message.py:80]: Add `:rtype` i think

File: interactions/api/models/message.py [hunk 3]
```diff
@@ -708,6 +718,15 @@ class Message(ClientSerializerMixin, IDMixin):
     )  # deprecated
     position: Optional[int] = field(default=None, repr=False)
 
+    @property
+    def deletable(self) -> bool:
```
Comment (Damego) [inline interactions/api/models/message.py:722]: Add `:rtype` i think

**Review by Damego — APPROVED** [review]

**Comment by Jonpot** [comment]
Hm. Seems like this didn't fix the bug. I'm still encountering ratelimits when the bot tries to delete undeletable messages. Version 4.3.4

**Comment by Damego** [comment]
> Hm. Seems like this didn't fix the bug. I'm still encountering ratelimits when the bot tries to delete undeletable messages. Version 4.3.4

v4.3.4 was released in october, 26. This pr was merged november, 26. How it's supposed to appears in v4.3.4? You can install v4.4.0 beta or wait for new release.

**Comment by Jonpot** [comment]
Whoops. I admittedly didn't check, I just assumed that a 3-month old PR would have made it to main by now. My bad. Used a lambda to get around it in the meantime.

**Pull request merged**